airflow/executors/kubernetes_executor.py

```diff
@@ -50,8 +50,17 @@
 from airflow.utils.session import provide_session
 from airflow.utils.state import State
 
-# TaskInstance key, command, configuration, pod_template_file
-KubernetesJobType = Tuple[TaskInstanceKey, CommandType, Any, Optional[str]]
+
+class KubernetesJobType(NamedTuple):
+    key: TaskInstanceKey
+    command: CommandType
+    config: Any
+    pod_template_file: Optional[str]
+    next_allowed_retry: datetime
+    retry_count: int
+
+
+RETRY_BACKOFF_SECONDS = [1, 2, 5, 10, 10, 30, 60, 60]
```

Comment:
Are these numbers reasonable? Or should it continue longer?

Reply:
As a counterpoint, perhaps a formula for the backoff (exponential backoff? doubling?) and set a limit to the number of retries instead of defining the exact times like this?

Reply:
+1 to a formula (e.g. exponential but no longer than 60 seconds)